godot-kotlin/godot-library/src/nativeCore/kotlin/godot/core/type/Color.kt

```diff
@@ -0,0 +1,806 @@
+     */
+    fun toABGR64(): Int = toByteColorCode(a, b, g, r, ColorByte.BITS64)
+
+    /**
+     * Returns the color’s 32-bit integer in ARGB format (each byte represents a component of the ARGB profile).
+     * ARGB is more compatible with DirectX.
+     */
+    fun toARGB32(): Int = toByteColorCode(a, r, g, b, ColorByte.BITS32)
+
+    /**
+     * Returns the color’s 64-bit integer in ARGB format (each word represents a component of the ARGB profile).
+     * ARGB is more compatible with DirectX.
+     */
+    fun toARGB64(): Int = toByteColorCode(a, r, g, b, ColorByte.BITS64)
+
+    /**
+     * Returns the color’s 32-bit integer in RGBA format (each byte represents a component of the RGBA profile).
+     * RGBA is Godot’s default format.
+     */
+    fun toRGBA32(): Int = toByteColorCode(r, g, b, a, ColorByte.BITS32)
+
+    /**
+    Returns the color’s 64-bit integer in RGBA format (each word represents a component of the RGBA profile).
+    RGBA is Godot’s default format.
+     */
+    fun toRGBA64(): Int = toByteColorCode(r, g, b, a, ColorByte.BITS64)
+
+    private fun toByteColorCode(first: RealT, second: RealT, third: RealT, fourth: RealT, colorByte: ColorByte): Int {
+        val size = colorByte.size
+        val shift = colorByte.shift
+        var c: Int = (first * size).roundToInt()
+        c = c shl shift
+        c = c or (second * size).roundToInt()
+        c = c shl shift
+        c = c or (third * size).roundToInt()
+        c = c shl shift
+        c = c or (fourth * size).roundToInt()
+
+        return c
+    }
+
+    /**
+     * Returns the color’s HTML hexadecimal color string in ARGB format (ex: ff34f822).
+     * Setting with_alpha to false excludes alpha from the hexadecimal string.
+     */
+    fun toHtml(alpha: Boolean): String {
+        var txt = ""
+        txt += toHex(r)
+        txt += toHex(g)
+        txt += toHex(b)
+        if (alpha) {
+            txt = toHex(a) + txt
+        }
+        return txt
+    }
+
+    private fun toHex(p_val: RealT): String {
+        var v = (p_val * 255).toInt()
+        v = when {
+            v < 0 -> 0
+            v > 255 -> 255
+            else -> v
+        }
+        var ret = ""
+
+        for (i in 0..1) {
+            val c = shortArrayOf(0, 0)
+            val lv = v and 0xF
+            if (lv < 10) {
+                c[0] = ('0'.toInt() + lv).toShort()
+            } else {
+                c[0] = ('a'.toInt() + lv - 10).toShort()
+            }
+
+            v = v shr 4
+            val cs = String(charArrayOf(c[0].toChar()))
+            ret = cs + ret
+        }
+        return ret
+    }
+
+    //Utilities
+    operator fun plus(c: Color) = Color(r + c.r, g + c.g, b + c.b, a + c.a)
+
+    operator fun minus(c: Color) = Color(r - c.r, g - c.g, b - c.b, a - c.a)
+
+    operator fun times(c: Color) = Color(r * c.r, g * c.g, b * c.b, a * c.a)
+
+    operator fun div(c: Color) = Color(r / c.r, g / c.g, b / c.b, a / c.a)
```

Comment:
Should we implement operators with real_t right operand, like here : https://github.com/godotengine/godot/blob/825a89d0dad7e69eabd5c431273e88a1ee74a75d/core/color.cpp#L585 ?